File: queue/src/main/java/br/com/queue/dtos/serviceManagement/statistics/ResponseServiceDashBoardDto.java
package br.com.queue.dtos.serviceManagement.statistics;

import java.util.List;

public record ResponseServiceDashBoardDto(

        ResponseCountTotalServicesStatisticsDto countTotalServicesStatistics,
        ResponseServicePercentagesStatisticsDto servicePercentagesStatistics,
        List<ResponseServicesCreatedByMonthStatisticsDto> servicesCreatedByMonth,
        List<ResponseServicesByDepartmentStatisticsDto> servicesByDepartment,
        List<ResponseUsersByServiceStatisticsDto> usersByService,
        List<ResponseSchedulesByServiceStatisticsDto> schedulesByService,
        List<ResponseTicketsByServiceStatisticsDto> ticketsByService
) {
}

File: queue-frontend/src/app/dtos/services/statistics/ResponseServiceDashBoardDto.ts
import { ResponseSchedulesByServiceStatisticsDto } from './ResponseSchedulesByServiceStatisticsDto';
import { ResponseCountTotalServicesStatisticsDto } from "./ResponseCountTotalServicesStatisticsDto"
import { ResponseServicePercentagesStatisticsDto } from "./ResponseServicePercentagesStatisticsDto";
import { ResponseServicesByDepartmentStatisticsDto } from "./ResponseServicesByDepartmentStatisticsDto";
import { ResponseServicesCreatedByMonthStatisticsDto } from "./ResponseServicesCreatedByMonthStatisticsDto";
import { ResponseTicketsByServiceStatisticsDto } from "./ResponseTicketsByServiceStatisticsDto";
import { ResponseUsersByServiceStatisticsDto } from "./ResponseUsersByServiceStatisticsDto";


export interface ResponseServiceDashBoardDto {

  countTotalServicesStatistics: ResponseCountTotalServicesStatisticsDto;
  servicePercentagesStatistics: ResponseServicePercentagesStatisticsDto;
  servicesCreatedByMonth: ResponseServicesCreatedByMonthStatisticsDto[];
  servicesByDepartment: ResponseServicesByDepartmentStatisticsDto[];
  usersByService: ResponseUsersByServiceStatisticsDto[];
  schedulesByService: ResponseSchedulesByServiceStatisticsDto[];
  ticketsByService: ResponseTicketsByServiceStatisticsDto[]
}
